### PR TITLE
multi: error responses should not include the client's junk

### PR DESCRIPTION
**Pull request by chappjc**

### CODE REVIEW

File: server/auth/registrar.go [hunk 1]
```diff
@@ -214,7 +216,7 @@ func (auth *AuthManager) validateFee(conn comms.Link, acctID account.AccountID,
 	if err != nil {
 		msgErr = &msgjson.Error{
 			Code:    msgjson.RPCInternalError,
-			Message: "wrong fee address. wanted " + regAddr + " got " + addr,
+			Message: "wrong fee address. wanted " + regAddr,
```
Comment (buck54321) [inline server/auth/registrar.go:219]: Existing issue, but this message is wrong.

Comment (buck54321) [reply server/auth/registrar.go:219]: I think we'll want to see this error in the server logs.

File: server/auth/registrar.go [hunk 2]
```diff
@@ -57,9 +57,10 @@ func (auth *AuthManager) handleRegister(conn comms.Link, msg *msgjson.Message) *
 	// Register account and get a fee payment address.
 	feeAddr, err := auth.storage.CreateAccount(acct)
 	if err != nil {
+		log.Debugf("CreateAccount(%v) failed: %v", acct, err)
```
Comment (buck54321) [inline server/auth/registrar.go:60]: Maybe a `log.Infof`?

File: server/auth/registrar.go [hunk 3]
```diff
@@ -155,9 +156,10 @@ func (auth *AuthManager) handleNotifyFee(conn comms.Link, msg *msgjson.Message)
 	// Get the registration fee address assigned to the client's account.
 	regAddr, err := auth.storage.AccountRegAddr(acctID)
 	if err != nil {
+		log.Debugf("AccountRegAddr failed to load info for account %v: %v", acctID, err)
```
Comment (buck54321) [inline server/auth/registrar.go:159]: Maybe `log.Infof`?

**Review by buck54321 — CHANGES_REQUESTED** [review]
Are we certain that the `Unmarshal` errors are not worth logging on the server?

File: server/auth/registrar.go (hunk 2 above)
Comment (chappjc) [reply server/auth/registrar.go:60]: This one can be hit by the user at will (acct already exists in table).  Ideally we'd log differently for legit errors though.

**Comment by chappjc** [comment]
> Are we certain that the `Unmarshal` errors are not worth logging on the server?

Fairly certain.  It indicates the client's payload was bad, and the error usually has a bits of that data if not all of it, so dumping it to the log is generally not desirable

**Review by buck54321 — APPROVED** [review]

**Pull request merged**